### PR TITLE
Show a setup screen when project has no commits

### LINKED ISSUE #3057: Onboarding feedback
Hey, I took another shot at onboarding and it's getting smoother each time. I found a couple minor issues:

1. If starting from a fresh Git repo, experiments will break if there's not a commit already. Is there some way we can auto commit when initializing dvc or suggest to make a commit if there is none in the repo? IMO this should be part of the "Show Setup" workflow, and we should block any other workflows until there is a commit in the repo.
2. The first thing I see in the sidebar after initializing my repo but before running any experiments is "Columns" showing the column "Created." Can we move it down in the sidebar or hide it?

--------

Actions:

- [x] Check if this is about `dvc exp run` and `dvc exp show` in the first place. If they return a reasonable error we can and should show them.
- [ ] If there are not commits we can introduce a screen with a message and a button to focus users on the SCM panel.
- [x] Disable columns section (and other sections) if we can't show the table at this moment yet to make it consistent.

### PR DESCRIPTION
**Pull request by julieg18**
* adds an extra step to our setup that shows if a user has initialized git and dvc but still has no commits
* takes off notification that appears after dvc initalizes

## Demo

https://user-images.githubusercontent.com/43496356/217660720-0d7b5ff3-7034-4124-a943-4703fc8b48ed.mov

Followup to #3239, Fixes #3057

### CODE REVIEW

File: webview/src/setup/components/NeedsGitCommit.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { Button } from '../../shared/components/button/Button'
+
+type NeedsGitCommitProps = { showScmPanel: () => void }
+
+export const NeedsGitCommit: React.FC<NeedsGitCommitProps> = ({
+  showScmPanel
+}) => (
+  <EmptyState>
+    <div>
+      <h1>Your project has no commits in Git</h1>
```
Comment (julieg18) [inline webview/src/setup/components/NeedsGitCommit.tsx:12]: Could any of these statements be improved?

Comment (mattseddon) [inline webview/src/setup/components/NeedsGitCommit.tsx:12]: [C] No Git commits detected

File: webview/src/setup/components/NeedsGitCommit.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      <p>
+        DVC requires your project to have atleast one commit to run experiments
+        correctly.{' '}
```
Comment (mattseddon) [inline webview/src/setup/components/NeedsGitCommit.tsx:15]: [C] At least one commit is required to enable experiments.

**Review by mattseddon — APPROVED** [review]

File: webview/src/setup/components/NeedsGitCommit.tsx (hunk 2 above)
Comment (sroy3) [inline webview/src/setup/components/NeedsGitCommit.tsx:15]: That last space is probably uneccessary.

**Review by sroy3 — APPROVED** [review]

File: webview/src/setup/components/NeedsGitCommit.tsx (hunk 2 above)
Comment (julieg18) [reply webview/src/setup/components/NeedsGitCommit.tsx]: Nice catch!

File: webview/src/setup/components/NeedsGitCommit.tsx (hunk 1 above)
Comment (julieg18) [reply webview/src/setup/components/NeedsGitCommit.tsx]: Thanks!

**Pull request merged**